core/metacling/src/TCling.cxx

```diff
@@ -3574,10 +3574,14 @@ void TCling::SetClassInfo(TClass* cl, Bool_t reload)
    cl->fClassInfo = 0;
    std::string name(cl->GetName());
 
+   bool useCxxModules = false;
+#ifdef R__USE_CXXMODULES
+   useCxxModules = true;
+#endif
    // Handle the special case of 'tuple' where we ignore the real implementation
    // details and just overlay a 'simpler'/'simplistic' version that is easy
    // for the I/O to understand and handle.
-   if (strncmp(cl->GetName(),"tuple<",strlen("tuple<"))==0) {
+   if (!(useCxxModules && IsFromRootCling()) && strncmp(cl->GetName(),"tuple<",strlen("tuple<"))==0) {
```

Comment:
If the solution is not to store the emulated tuple in the pcm I’d be much more comfortable with this PR if we do not store it in the pch too. That’d mean dropping the useCxxModules variable.

Reply:
If we remove 'useCxxModule' this means that without the module (i.e. current default) the alternate information will not be stored the rootpcm, this will lead to auto-parsing at run-time which would be a disaster.  [Even with the module, I am not clear whether we 'need' the info put into the rootpcm ... i.e. we would need to measure what is the cost]